src/clj/puppetlabs/trapperkeeper/services/metrics/jolokia.clj

```diff
@@ -73,11 +75,29 @@
 
 (defn create-servlet
   "Builds a Jolokia Servlet that uses Clojure logging."
-  []
-  (proxy [AgentServlet] []
-    ;; NOTE: An alternative to this method override would be to use defrecord
-    ;; to create a class that can be set as `:logHandlerClass` in the servlet
-    ;; configuration. This requires AOT compilation for the namespace defining
-    ;; the record so that Jolokia can find the resulting class.
-    (createLogHandler [_ _]
-      (create-logger))))
+  [auth-check-fn]
+  (let [check-auth (if auth-check-fn
+                     (fn [^HttpServletRequest request]
+                      (auth-check-fn
+                       (ring-servlet/build-request-map request)))
+                     (fn [^HttpServletRequest request]
+                       {:authorized true :message ""}))]
+    (if auth-check-fn
+      (log/info "Metrics access control using trapperkeeper-authorization is enabled.")
+      (log/warn "Metrics access control using trapperkeeper-authorization is disabled. Add the authorization service to the trapperkeeper bootstrap configuration file to enable it."))
+    (proxy [AgentServlet] []
+     ;; NOTE: An alternative to this method override would be to use defrecord
+     ;; to create a class that can be set as `:logHandlerClass` in the servlet
+     ;; configuration. This requires AOT compilation for the namespace defining
+     ;; the record so that Jolokia can find the resulting class.
+     (createLogHandler [_ _]
+       (create-logger))
+     (service [request response]
```

Comment:
Just to make sure I'm understanding the clojure here, this adds an additional method override on `AgentServlet` for the service function? Where as previously it used a default definition (that we patch through to with `proxy-super`)?

Reply:
Yes. `proxy-super` is equivalent to calling `super` in java